### ISSUE TITLE
LICENSE.txt missing for several releases on https://releases.llvm.org/

### ISSUE THREAD

**Issue by swaldhoer** [llvm/llvm-project#57672]
The license information is inconsistently available on https://releases.llvm.org/. This comes from the ``release`` directoy not beeing consistent layouted.
See the following examples for inconsitency of the releases:

- LLVM 13.0.0:
  - release directory https://releases.llvm.org/13.0.0/ exists ✅
  - license information https://releases.llvm.org/13.0.0/LICENSE.TXT exists ✅
- LLVM 13.0.1:
  - release directory https://releases.llvm.org/13.0.1/ exists ✅
  - license information https://releases.llvm.org/13.0.1/LICENSE.TXT exists ✅
- LLVM 14.0.0:
  - release directory https://releases.llvm.org/14.0.0/ exists ✅
  - license information https://releases.llvm.org/14.0.0/LICENSE.TXT exists ✅
- LLVM 14.0.1: (it's the same for all 14.0.x releases including latest 14.0.6)
  - release directory https://releases.llvm.org/14.0.1/ does not exist ❌
  - license information https://releases.llvm.org/14.0.1/LICENSE.TXT does not exists ❌
- LLVM 15.0.0:
  - release directory https://releases.llvm.org/15.0.0/ does not exist ✅
  - license information https://releases.llvm.org/15.0.0/LICENSE.TXT does not exists ❌

I have seen that the LLVM release process already states that the license file should be copied to the release directory (https://releases.llvm.org/15.0.0/docs/HowToReleaseLLVM.html#update-the-llvm-website)

Is there a reason, that it has not been done for some releases? If this ws just forgotten, could the licensing information be added back?

**Comment by DimitryAndric** [comment]
The documentation is likely out of date. Releases are now published right here on GitHub: https://github.com/llvm/llvm-project/releases

(ping @tstellar @tru)

**Comment by DimitryAndric** [comment]
(Btw, you are still correct in that LICENSE.txt is not explicitly put under every single release.)

**Comment by tru** [comment]
This is probably just missed and needs to be automated and not a manual process. But I am more keen on just pointing links to the GitHub release tags LICENSE.txt instead of copying the file.

Is there a reason you look for the files right there or if just links to the tag in the repo?

**Comment by swaldhoer** [comment]
> The documentation is likely out of date. Releases are now published right here on GitHub: https://github.com/llvm/llvm-project/releases
> 
> (ping @tstellar @tru)

The GitHub release page does not provide licensing information for each release.

> This is probably just missed and needs to be automated and not a manual process. But I am more keen on just pointing links to the GitHub release tags LICENSE.txt instead of copying the file.
> 
> Is there a reason you look for the files right there or if just links to the tag in the repo?

Yes, because this was the only and reliable way of getting licensing information for every release. There was no single licensing information on the main branch prior to #57332. And as this is a recent change, there is no release branch or tagged release, that has a license file you could point to, therefore using https://releases.llvm.org/ was the solution.

**Comment by tru** [comment]
This works: https://github.com/llvm/llvm-project/blob/llvmorg-14.0.0/llvm/LICENSE.TXT

**Comment by swaldhoer** [comment]
> This works: https://github.com/llvm/llvm-project/blob/llvmorg-14.0.0/llvm/LICENSE.TXT

Yes it does, but is this a future proof solution? I like having the license information on the project website more TBH.

**Comment by tru** [comment]
@kbeyls any thoughts on this?

**Comment by kbeyls** [comment]
> @kbeyls any thoughts on this?

I'm afraid I don't have constructive thoughts on this beyond that it should be clear for every release what the license(s) is/are that apply.
I would assume that making sure the license.txt file is present in all packages that can be downloaded from the release page would be sufficient, but I'm open to hearing why that wouldn't be the case and what in addition would be helpful.

I'm guessing that maybe in time, we'll end up implementing some form of SPDX package information [(https://spdx.github.io/spdx-spec/scope/](https://github.com/david-a-wheeler/spdx-tutorial#spdx-files) if these become more popular and used more. But I'd say that is obviously an orthogonal issue to this issue.

**Comment by swaldhoer** [comment]
As of now, the release `15.0.1` to `15.0.4` happend, but for none of them the `LICENSE.txt` file is available on the server.

However, in the meanwhile, `LICENSE.txt` for 15.0.0 is now [available}(https://releases.llvm.org/15.0.0/LICENSE.TXT).

**Comment by tru** [comment]
This is correct - we don't have a automated way to do this yet, I worked on a solution for a new releases page where this would be solved. But that has been stuck on some infrastructure work. I hope we can move forward with it soon.

**Comment by swaldhoer** [comment]
> As of now, the release `15.0.1` to `15.0.4` happend, but for none of them the `LICENSE.txt` file is available on the server.
> 
> However, in the meanwhile, `LICENSE.txt` for 15.0.0 is now [available](https://releases.llvm.org/15.0.0/LICENSE.TXT).

@tru the link is broken again. Why has this been changed? I mean, soneone must have actively removed it😅

Is there a common understanding how this will be handled in die future?
